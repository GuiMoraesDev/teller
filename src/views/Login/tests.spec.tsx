import React from 'react';

import userEvent from '@testing-library/user-event';

import { render, waitFor } from 'tests/test.utils';

import LoginTemplate from './template';

const ContainerRender = (mockSubmit: jest.Mock<any, any>) =>
	render(<LoginTemplate onSubmit={mockSubmit} isLoading={false} />);

const elementsTestId = {
	submitButton: 'submit-button',
	usernameInput: "username-input"
};

describe('FloatSubMenu pattern', () => {
	it('should render correctly', () => {
		const mockSubmit = jest.fn();

		const { container } = ContainerRender(mockSubmit);

		expect(container.firstChild).toMatchSnapshot();
	});

	it('should not call submit form function without type a username', async () => {
		const mockSubmit = jest.fn();

		const { getByTestId } = ContainerRender(mockSubmit);

		const submitButton = getByTestId(elementsTestId.submitButton);

		await waitFor(() => userEvent.click(submitButton));

		expect(mockSubmit).toBeCalledTimes(0);
	});

	it('should call submit form function with username filled', async () => {
		const mockSubmit = jest.fn();

		const { getByTestId } = ContainerRender(mockSubmit);

		const submitButton = getByTestId(elementsTestId.submitButton);
		const usernameInput = getByTestId(elementsTestId.usernameInput);

		await waitFor(async () => {
			await userEvent.type(usernameInput, "my-username");
			await userEvent.click(submitButton);
		});

		expect(mockSubmit).toBeCalledTimes(1);
	});
});
