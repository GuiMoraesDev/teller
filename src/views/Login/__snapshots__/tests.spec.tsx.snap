// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatSubMenu pattern should render correctly 1`] = `
.c7 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 4.4rem;
  border: 0.1rem solid #C9CAC8;
  border-radius: 0.4rem;
  margin: 0;
  padding: 0 1.4rem;
  gap: 1.2rem;
  -webkit-transition: background-color 0.2s, color 0.2s;
  transition: background-color 0.2s, color 0.2s;
  cursor: pointer;
  color: #FFFFFF;
  background-color: #63989F;
  border-color: #4F7A7F;
}

.c7:hover {
  background-color: #82ADB2;
}

.c7 svg {
  color: inherit;
}

.c7:hover {
  background-color: #4F7A7F;
}

.c7 .spin {
  -webkit-animation-name: ieAXeN;
  animation-name: ieAXeN;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c1 {
  position: relative;
  padding: 0;
  margin: 0;
  font-size: 4.8rem;
  font-weight: 500;
  font-family: 'Montserrat',sans-serif;
  line-height: 8.0rem;
  color: #181917;
  box-sizing: border-box;
}

.c3 {
  position: relative;
  padding: 0;
  margin: 0;
  font-size: 1.6rem;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  line-height: 2.4rem;
  color: #181917;
  box-sizing: border-box;
}

.c2 {
  position: relative;
  width: 100%;
}

.c4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 4.4rem;
  color: #181917;
  background-color: #F2F2F1;
  border: 0.1rem solid #C9CAC8;
  border-radius: 0.4rem;
  margin: 0;
  padding: 0 1.4rem;
  gap: 1.2rem;
  -webkit-transition: box-shadow 0.2s, border-color 0.2s;
  transition: box-shadow 0.2s, border-color 0.2s;
}

.c4:hover,
.c4:focus-within {
  border-color: #797B75;
  box-shadow: 1px 0px 3px 0px #E4E5E3;
}

.c4 svg {
  color: #949591;
}

.c5 {
  outline: none;
  width: 100%;
  height: 100%;
  color: inherit;
  background-color: inherit;
  border: none;
  border-radius: inherit;
}

.c6 {
  position: absolute;
  bottom: 0.2rem;
  right: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  font-size: 1.2rem;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  line-height: 1.6rem;
  margin: auto;
  padding: 1.2rem;
  gap: 1.2rem;
  color: #FFFFFF;
  background-color: #EE977D;
  border-radius: none;
  border-bottom-left-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  opacity: 0;
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
  -webkit-transition: opacity 0.5s ease-in-out;
  transition: opacity 0.5s ease-in-out;
}

.c0 {
  grid-area: Content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 2.0rem;
}

<main
  class="c0"
>
  <strong
    class="c1"
  >
    Welcome to Teller
  </strong>
  <form
    id="github-login"
  >
    <div
      class="c2"
    >
      <label
        class=""
        for="github-user"
      >
        <p
          class="c3"
        >
          Type below your github user to login
        </p>
      </label>
      <label
        class="c4"
        for="github-user"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <path
            d="M84,240a23.9,23.9,0,0,0,24-24V168"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M172,240a23.9,23.9,0,0,1-24-24V168"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M152,168h16a23.9,23.9,0,0,1,24,24v8a23.9,23.9,0,0,0,24,24"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M104,168H88a23.9,23.9,0,0,0-24,24v8a23.9,23.9,0,0,1-24,24"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M111.8,64A52,52,0,0,0,68,40a52,52,0,0,0-3.5,44.7A49.3,49.3,0,0,0,56,112v8a48,48,0,0,0,48,48h48a48,48,0,0,0,48-48v-8a49.3,49.3,0,0,0-8.5-27.3A52,52,0,0,0,188,40a52,52,0,0,0-43.8,24Z"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
        </svg>
        <input
          class="c5"
          data-testid="username-input"
          id="github-user"
          name="username"
          placeholder="@my-github-username"
        />
      </label>
      <span
        class="c6"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <line
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
            x1="128"
            x2="128"
            y1="104"
            y2="144"
          />
          <path
            d="M114.2,40l-88,152A16,16,0,0,0,40,216H216a16,16,0,0,0,13.8-24l-88-152A15.9,15.9,0,0,0,114.2,40Z"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <circle
            cx="128"
            cy="180"
            r="12"
          />
        </svg>
      </span>
    </div>
  </form>
  <button
    class="c7"
    data-testid="submit-button"
    form="github-login"
    type="submit"
  >
    Login
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="none"
        height="256"
        width="256"
      />
      <polyline
        fill="none"
        points="94 170 136 128 94 86"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
      />
      <line
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
        x1="24"
        x2="136"
        y1="128"
        y2="128"
      />
      <path
        d="M136,40h56a8,8,0,0,1,8,8V208a8,8,0,0,1-8,8H136"
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
      />
    </svg>
  </button>
</main>
`;
